xbmc/Application.cpp

```diff
@@ -3450,7 +3450,15 @@ void CApplication::OnPlayBackSeekChapter(int iChapter)
 
 void CApplication::OnAVChange()
 {
+  CLog::LogF(LOGDEBUG, "CApplication::OnAVChange");
+
   CStereoscopicsManager::GetInstance().OnStreamChange();
+
+#ifdef HAS_PYTHON
+  // informs python script currently running playback has started
+  // (does nothing if python is not loaded)
+  g_pythonParser.OnAVChange();
```

Comment:
clients of this evant must not do much processing on this event. triggering gui actions can even lead to dead locks. 
shouldn't all python event be served by the main thread?

Reply:
Isn't CApplication the main thread? I might not be understanding you right. 
Basically this was a copy and paste job.

Reply:
> Isn't CApplication the main thread?

A class is not a thread. Those callbacks are called from a player thread. Have a look at onPlaybackStarted / stopped for example. They fire a gui message that is received by the main thread.